NICE.Identity.Management/ClientApp/src/views/ServicesList/ServicesList.tsx

```diff
@@ -0,0 +1,411 @@
+
+		const querystringObject = queryStringToObject(querystring);
+
+		const pageNumber = Number(
+			querystringObject.page
+				? Array.isArray(querystringObject.page)
+					? querystringObject.page[0]
+					: querystringObject.page
+				: 1,
+		);
+
+		let itemsPerPage = querystringObject.amount
+			? Array.isArray(querystringObject.amount)
+				? querystringObject.amount[0]
+				: querystringObject.amount
+			: 25;
+
+		itemsPerPage = Number(itemsPerPage) ? Number(itemsPerPage) : itemsPerPage;
+
+		this.state = {
+			path: "",
+			originalWebsites: [],
+			websites: [],
+			isLoading: true,
+			pageNumber: pageNumber,
+			itemsPerPage: itemsPerPage,
+			environmentFiltersChecked: [],
+			environmentsForFilter: [],
+		};
+
+		document.title = "NICE Accounts - Services list";
+	}
+
+	async componentDidMount(): Promise<void> {
+		this.setState({ isLoading: true });
+
+		const websites = await fetchData(Endpoints.websitesList);
+
+		if (isDataError(websites)) {
+			this.setState({ error: websites });
+		}else{
+		const allEnvironments = websites.map((website: { environment: { name: string; }; }) => website.environment.name);
+
+		const environmentsForFilter = allEnvironments.reduce(function (accumulatedEnvironments: string[], currentEnvironment: string) {
+			if (accumulatedEnvironments.indexOf(currentEnvironment) === -1) {
+				accumulatedEnvironments.push(currentEnvironment)
+			}
+			return accumulatedEnvironments
+		}, []);
+
+		this.setState({environmentsForFilter})
+		}
+
+		this.setState({ originalWebsites: websites, websites: websites, isLoading: false });
+	}
+
+	pastPageRange = (
+		itemsPerPage: string | number,
+		pageNumber: number,
+		dataCount: number,
+	): number => {
+		let pastPageRange = false;
+
+		if (Number(itemsPerPage)) {
+			itemsPerPage = Number(itemsPerPage);
+			pastPageRange = pageNumber * itemsPerPage >= dataCount;
+		}
+
+		if (pastPageRange || itemsPerPage === "all") {
+			pageNumber = 1;
+		}
+
+		return pageNumber;
+	};
+
+	filterWebsitesByEnvironment = (environment: string): void => {
+		this.setState({ isLoading: true });
+
+		let environmentFiltersChecked = this.state.environmentFiltersChecked;
+
+		environmentFiltersChecked = environmentFiltersChecked.includes(environment)
+			? environmentFiltersChecked.filter((environmentFilter) => environmentFilter !== environment)
+			: environmentFiltersChecked.concat(environment);
+
+		const itemsPerPage = Number(this.state.itemsPerPage)
+			? Number(this.state.itemsPerPage)
+			: this.state.itemsPerPage;
+
+		let websites = this.state.originalWebsites,
+			pageNumber = this.state.pageNumber;
+
+		if (environmentFiltersChecked.length)
+			websites = this.websitesByEnvironment(environmentFiltersChecked, websites);
+
+		pageNumber = this.pastPageRange(
+			itemsPerPage,
+			pageNumber,
+			this.state.websites.length,
+		);
+
+		this.setState({
+			websites: websites,
+			environmentFiltersChecked: environmentFiltersChecked,
```

Comment:
```suggestion
			websites,
			environmentFiltersChecked,
```

You can write this as above if the variables by the same name are available.